src/main/java/org/veupathdb/service/eda/ds/plugin/correlation/AbstractCorrelationBipartiteNetwork.java

```diff
@@ -113,6 +113,8 @@ protected void writeResults(OutputStream out, Map<String, InputStream> dataStrea
     uniqueNodeIDs.forEach((nodeID) -> {
       NodeData node = new NodeDataImpl();
       node.setId(nodeID);
+      int nodeDegree = Collections.frequency(nodeIDs, nodeID);
```

Comment:
worth noting that this only works because we've ensured the edge list is de-duplicated